### PR TITLE
beacon chain archival node

### PR DESCRIPTION
**Pull request by alajko**
## Issue

Create a Archival Node that syncs with and stores the beaconchain.

## TODO

- [x] integrate it with `deploy.sh`
- [x] Test archival node.
- [ ] Test archival node on AWS.

### CODE REVIEW

File: cmd/archival/main.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+	nodeConfig.P2pPriKey, _, err = utils.LoadKeyFromFile(*keyFile)
+	if err != nil {
+		panic(err)
+	}
+
+	// Initialize leveldb for main blockchain and beacon.
+	if nodeConfig.MainDB, err = InitLDBDatabase(*ip, *port, *freshDB, false); err != nil {
+		panic(err)
+	}
+	if nodeConfig.BeaconDB, err = InitLDBDatabase(*ip, *port, *freshDB, true); err != nil {
```
Comment (chaosma) [inline cmd/archival/main.go]: For archival node, we only need to store one DB depends on which shard the archival node belongs to

Comment (alajko) [reply cmd/archival/main.go]: Done

File: cmd/harmony/main.go [hunk 2]
```diff
@@ -82,6 +82,8 @@ var (
 	keyFile = flag.String("key", "./.hmykey", "the private key file of the harmony node")
 	// isBeacon indicates this node is a beacon chain node
 	isBeacon = flag.Bool("is_beacon", false, "true means this node is a beacon chain node")
+	// isArchival indicates this node is a archival node that will save and archive current blockchain
+	isArchival = flag.Bool("is_archival", true, "true means this node is a archival node")
```
Comment (LeoHChen) [inline cmd/harmony/main.go]: it has to be false by default.

File: cmd/harmony/main.go [hunk 3]
```diff
@@ -175,6 +177,39 @@ func createGlobalConfig() *nodeconfig.ConfigType {
 	return nodeConfig
 }
 
+func createArchivalNode() (*node.Node, *nodeconfig.ConfigType) { //Mix of setUpConsensusAndNode and createGlobalConfig
+	var err error
+	nodeConfig := nodeconfig.GetGlobalConfig()
+
+	// Currently we hardcode only one shard.
+	nodeConfig.ShardIDString = "10000" //ShardID can be highest positive number of shard.
```
Comment (LeoHChen) [inline cmd/harmony/main.go]: what's this about?  ArchivalNode will need to join beacon chain to start.

Comment (alajko) [reply cmd/harmony/main.go]: Done.

File: cmd/harmony/main.go (hunk 2 above)
Comment (alajko) [reply cmd/harmony/main.go]: Done.

**Comment by alajko** [comment]
Local Run for Archival Gist: https://gist.github.com/alajko/cab4371636b4d77cc343532aaf67a112
General Running Log: https://gist.github.com/alajko/d94f980ed083830c007db741dc211255

**Pull request closed without merging**